File: app/Models/JadwalMengajar.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class JadwalMengajar extends Model
{
    protected $table = 'jadwal_mengajar';
    public $timestamps = false;

    protected $fillable = [
        'uid',
        'hari',
        'mata_kuliah',
        'jam_mulai',
        'jam_selesai',
        'ruangan',
        'keterangan',
    ];

    public function dosen(): BelongsTo
    {
        return $this->belongsTo(Dosen::class, 'uid', 'uid');
    }
}

File: lib/features/jadwal/domain/jadwal_mengajar.dart
class JadwalMengajar {
  final int id;
  final String uid;
  final String hari;
  final String mataKuliah;
  final String namaKelas;
  final String jamMulai;
  final String jamSelesai;
  final String ruangan;
  final String keterangan;

  const JadwalMengajar({
    this.id = 0,
    required this.uid,
    required this.hari,
    required this.mataKuliah,
    this.namaKelas = '',
    required this.jamMulai,
    required this.jamSelesai,
    this.ruangan = '',
    this.keterangan = '',
  });

  String get label => namaKelas.isEmpty ? mataKuliah : '$mataKuliah - Kelas $namaKelas';

  factory JadwalMengajar.fromMap(Map<String, dynamic> map) => JadwalMengajar(
        id: map['id'] is int ? map['id'] as int : int.tryParse('${map['id']}') ?? 0,
        uid: map['uid'] ?? '',
        hari: map['hari'] ?? '',
        mataKuliah: map['mata_kuliah'] ?? '',
        namaKelas: map['nama_kelas'] ?? '',
        jamMulai: map['jam_mulai'] ?? '',
        jamSelesai: map['jam_selesai'] ?? '',
        ruangan: map['ruangan'] ?? '',
        keterangan: map['keterangan'] ?? '',
      );

  Map<String, dynamic> toMap() => {
        'uid': uid,
        'hari': hari,
        'mata_kuliah': mataKuliah,
        'nama_kelas': namaKelas,
        'jam_mulai': jamMulai,
        'jam_selesai': jamSelesai,
        'ruangan': ruangan,
        'keterangan': keterangan,
      };
}
